main.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	"net/http"
+	"os"
+	"strings"
+
+	"k8s.io/api/admission/v1beta1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func isResourceSet(resList corev1.ResourceList, name corev1.ResourceName) bool {
+	var missing = resList == nil
+	if !missing {
+		if _, ok := resList[name]; !ok {
+			missing = true
+		}
+	}
+	return !missing
+}
+
+func validate(ar v1beta1.AdmissionReview, config *config) *v1beta1.AdmissionResponse {
+	reviewResponse := v1beta1.AdmissionResponse{}
+
+	var validationMsg string
+	reviewResponse.Allowed = true
+
+	deserializer := codecs.UniversalDeserializer()
+
+	if ar.Request.Kind.Kind == "Deployment" {
+		deployment := appsv1.Deployment{}
+
+		raw := ar.Request.Object.Raw
+		if _, _, err := deserializer.Decode(raw, nil, &deployment); err != nil {
+			log.Error(err)
+			return toAdmissionResponse(err)
+		}
+
+		log.Debugf("Admitting deployment: %+v", deployment)
+
+		var containerMsg string
+		checkResourceIsSet := func(resList corev1.ResourceList, name corev1.ResourceName, enabled bool, message string) {
+			if enabled && !isResourceSet(resList, name) {
+				containerMsg = containerMsg + message + " "
+			}
+		}
+
+		for _, container := range deployment.Spec.Template.Spec.Containers {
+			containerMsg = ""
```

Comment:
uh, can't you pass containerMsg as parameter to `checkResourceIsSet `? it's just kind of ugly for me that the function uses a field but it does not get that as a dependency

Reply:
yeah, I've butchered this piece with some last-minute fixes, it no longer makes much sense.. will fix it ;)